Cc 21 display menu options

AshokkSS opened the pull request:
Completed ticket as described.
Menu options are clearly displayed.

Commit:
Changed the structure of the main run file

Still needs tests done

Commit:
Completed.

All options are displayed clearly.

Commit:
cleared run_spec

Dark-Raine commented on .results.json:
```diff
@@ -0,0 +1 @@
+{"version":"3.11.0","messages":["\nAn error occurred while loading ./spec/run_spec.rb.\n\u001b[31mFailure/Error: \u001b[0m\u001b[1;32m$result\u001b[0m = \u001b[1;34;4mPROMPT\u001b[0m.select(\u001b[31m\u001b[1;31m\"\u001b[0m\u001b[31mPlease select an option below.\u001b[1;31m\"\u001b[0m\u001b[31m\u001b[0m, \u001b[1;34;4mOPTIONS\u001b[0m)\u001b[0m\n\u001b[31m\u001b[0m\n\u001b[31mNoMethodError:\u001 ...
```
Do we need to store this file in git?

Dark-Raine commented on lib/run.rb:
```diff
@@ -1,34 +1,101 @@
 CLEAR = "\e[H\e[2J"
+PROMPT = TTY::Prompt.new
+OPTIONS =  %w(Price_Check Compare_Cryptos My_Portfolio Close)
+PING_URL = "https://api.coingecko.com/api/v3/ping"
```
Could this live in it's own class where it's domain is just to do with connecting with the poke api?

Dark-Raine commented on lib/run.rb:
```diff
@@ -1,34 +1,101 @@
 require_relative '../config/environment'
 CLEAR = "\e[H\e[2J"
+PROMPT = TTY::Prompt.new
+OPTIONS =  %w(Price_Check Compare_Cryptos My_Portfolio Close)
```
Could these options not be bart of the main menu class?

Dark-Raine commented on lib/run.rb:
```diff
@@ -1,34 +1,101 @@
-    
-    
     
+    def ping_api
+        uri = URI(PING_URL)
+        res = Net::HTTP.get_response(uri)
+        if res.is_a?(Net::HTTPSuccess)
+            connected_box_success
+        else
+            connected_box_failure
+        end
+    end
```
This could be apart of the poke api class mentioned above

Dark-Raine commented on lib/run.rb:
```diff
@@ -1,34 +1,101 @@
+        end
+    end
+
+    def connected_box_success
+        box = TTY::Box.success("Connected to API")
+        print box
+        sleep 2
+        puts CLEAR
+    end
+
+    def connected_box_failure
+        box = TTY::Box.error("Unable to connect to API, Please close and reopen.")
+        print box
+        sleep 2
+        puts CLEAR
+    end    
```
Same thing for these

Dark-Raine commented on lib/run.rb:
```diff
@@ -1,34 +1,101 @@
+
+end
+
+# def capture_stdout(&blk)
+#     old = $stdout
+#     $stdout = fake = StringIO.new
+#     blk.call
+#     fake.string
+#   ensure
+#     $stdout = old
+#   end
```
delete commented code

Dark-Raine commented on lib/run.rb:
```diff
@@ -1,34 +1,101 @@
+#   ensure
+#     $stdout = old
+#   end
+
+
+
+
+
+
+
+
+
+
```
remove whitespace from files

Dark-Raine commented on spec/run_spec.rb:
```diff
@@ -0,0 +1,32 @@
+
+
+
+# describe "#loading_menu" do
+#   # expect{display_main_menu}.to eq('my message').to_stdout
+#   # printed = capture_stdout do
+#   #     display_main_menu
+#   #   end
+#     it 'displays menu options' do
+#       #test_prompt = TTY::Prompt.new
+      
+#       ping 
+#         #expect(PROMPT).to receive(:select).with(%w(Price_Check Compare_Cryptos My_Portfolio Close))
+#       expect(Net::HTTPSuccess).to receive(ping).and_return("correct")
+
+      
+    
+      
+#   end
+# end
+# context "#display_main_menu" do
+#   it 'should print "my message"' do
+#     printed = capture_stdout do
+#       display_main_menu # do your actual method call
+#     end
+
+#     printed.should eq("my message")
+#   end
+# end
```
delete commented out code

Dark-Raine commented:
Another thing to add, you haven't commited your gemfile and gemfile.lock. These are important in ensuring you always use the correct versions of the gems in your application

Commit:
Made changes from the comments.

Commit:
Update .gitignore

Commit:
Update .gitignore

Commit:
Update run.rb

AshokkSS commented:
Have made those changes now, just a few things.
I have added *.json to the gitignore but can still see it in my repo.
I can see the gem file is in the git?

Commit:
moved calls for crypto api to seperate file.

api.rb will be used for all api calls

Commit:
Update run.rb

Commit:
reformatted a lot of files. started writing tests.

Commit:
Moved Main menu class to own file, tidied up

Moved Main menu class to own file, wrote tests for pinging api, tidied up cryptoapi.rb

Commit:
removed .results.json

Commit:
Merge branch 'CC-4' into cc-21-display-menu-options

Commit:
Fixed gemfile
